test_common/harness/testHarness.c

```diff
@@ -655,23 +658,29 @@ int callTestFunctions( test_definition testList[], unsigned char selectedTestLis
                        cl_command_queue_properties queueProps )
 {
     int totalErrors = 0;
+    const Version device_version = get_device_cl_version(deviceToUse);
 
     for( int i = 0; i < testNum; ++i )
     {
         if( selectedTestList[i] )
         {
-            // Skip unimplemented test (can happen when you select all of the tests)
-            if( testList[i].func != NULL )
+            if (testList[i].min_version > device_version)
             {
-                int errors = callSingleTestFunction( testList[i], deviceToUse, forceNoContextCreation,
-                                                     numElementsToUse, queueProps );
-                resultTestList[i] = errors;
-                totalErrors += errors;
+                log_info("Skipping %s. Device doesn't support functionality exercised by this test.\n",
+                         testList[i].name);
+                continue;
             }
-            else
+
+            if (testList[i].func == NULL)
             {
                 log_info( "Skipping %s. Test currently not implemented.\n", testList[i].name );
+                continue;
```

Comment:
Results do not print test result status. It looks strange:
./test_conformance_device_timer CL_DEVICE_TYPE_GPU
 Initializing random seed to 0.
Requesting GPU device based on command line for platform index 0 and device index 0
Compute Device Name = Intel(R) Gen9 HD Graphics NEO, Compute Device Vendor = Intel(R) Corporation, Compute Device Version = OpenCL 2.0 NEO , CL C Version = OpenCL C 2.0
Supports single precision denormals: YES
sizeof( void*) = 8  (host)
sizeof( void*) = 8  (device)
Skipping timer_resolution_queries. Device doesn't support functionality exercised by this test.
Skipping device_and_host_timers. Device doesn't support functionality exercised by this test.

Reply:
Will rework.